### ISSUE TITLE
Rule Change: Make `no-nested-ternary` rule TypeScript syntax aware

### ISSUE THREAD

**Issue by lumirlumir** [eslint/eslint#19884]
### What rule do you want to change?

`no-nested-ternary`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint no-nested-ternary: "error" */

// JavaScript code with nested ternary operators

const thing = foo ? bar : baz === qux ? quxx : foobar;
foo ? (baz === qux ? quxx() : foobar()) : bar();

// TypeScript code with nested ternary operators

type KindOf<T> = T extends string
  ? 'string'
  : T extends number
    ? 'number'
    : T extends boolean
      ? 'boolean'
      : T extends any[]
        ? 'array'
        : 'other';
```

### What does the rule currently do for this code?

The `no-nested-ternary` rule currently doesn’t understand TypeScript syntax, so it fails to flag errors when nested  [conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) are used.

![Image](https://github.com/user-attachments/assets/0e0e846b-a244-4aef-99b8-01cb208885c7)

That’s because the TypeScript AST differs slightly from the JavaScript AST for ternary operations.

[TypeScript AST](https://typescript-eslint.io/play#ts=5.8.3&showAST=es&fileType=.tsx&code=C4TwDgpgBA0glgOwCYHkBmAeAKgPigXiiyggA9gJkBnKK4AJ0QHMBYAKCigH4oByOxgia92nAFxES5SkhoIArgFsARhHqjufBSrUiOUCcTIVqUZQHtzAGwgBDBBp68L1uwj3jJxmTXsgA2gC6jny29PS2IB4GfObAABa6ANzsQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

![Image](https://github.com/user-attachments/assets/c6b25fc9-5955-493c-ac7b-2f6ea99b4172)

### What will the rule do after it's changed?

For the reasons mentioned above, I’d like to request adding TypeScript syntax support to the `no-nested-ternary` rule.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

This issue is follow-up to https://github.com/eslint/eslint/issues/19173

**Comment by zaewc** [comment]
Hi! I totally agree with this issue.

Right now, the no-nested-ternary rule only checks for nested ternaries in regular JavaScript code (ConditionalExpression), so it ends up missing nested ternaries inside TypeScript’s conditional types.

That’s because ESLint is originally made for JavaScript, so it doesn’t really get TypeScript-specific stuff—unless you use something like `@typescript-eslint/parser` to help it understand.

**Comment by lumirlumir** [comment]
Ping @eslint/eslint-team

**Comment by nzakas** [comment]
Makes sense to me. :+1: Marking as accepted.

**Comment by kirkwaiblinger** [comment]
Can this be implemented with an option for the TS ternaries? Nested ternaries for conditional types is idiomatic, and (more importantly) doesn't really have a straightforward alternative the way non-type code does

**Comment by lumirlumir** [comment]
@kirkwaiblinger Sure, using an option makes sense to me 👍

Do you have any suggestions for the option name? For example, something like `allowConditionalTypes`?

**Comment by kirkwaiblinger** [comment]
That sounds good to me 👍

**Comment by nzakas** [comment]
Per https://github.com/eslint/eslint/pull/19908#discussion_r2200862917, we've agreed not to move forward with this change.
